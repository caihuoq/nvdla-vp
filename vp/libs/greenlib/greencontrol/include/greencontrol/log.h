//   GreenControl framework
//
// LICENSETEXT
//
//   Copyright (C) 2009 : GreenSocs Ltd
// 	 http://www.greensocs.com/ , email: info@greensocs.com
//
//   Developed by :
//   
//   Michael Ruetz <m.ruetz@tu-bs.de>,
//   Christian Schroeder <schroeder@eis.cs.tu-bs.de>
//     Technical University of Braunschweig, Dept. E.I.S.
//     http://www.eis.cs.tu-bs.de
//
//
//   This program is free software.
// 
//   If you have no applicable agreement with GreenSocs Ltd, this software
//   is licensed to you, and you can redistribute it and/or modify
//   it under the terms of the GNU General Public License as published by
//   the Free Software Foundation; either version 2 of the License, or
//   (at your option) any later version.
// 
//   If you have a applicable agreement with GreenSocs Ltd, the terms of that
//   agreement prevail.
// 
//   This program is distributed in the hope that it will be useful,
//   but WITHOUT ANY WARRANTY; without even the implied warranty of
//   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//   GNU General Public License for more details.
// 
//   You should have received a copy of the GNU General Public License
//   along with this program; if not, write to the Free Software
//   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
//   02110-1301  USA 
// 
// ENDLICENSETEXT

// doxygen comments

#ifndef __LOG_H__
#define __LOG_H__

#include "greencontrol/gclog/gc_logger.h"
#include "greencontrol/gclog/filter.h"
#include "greencontrol/gclog/output.h"

#endif
