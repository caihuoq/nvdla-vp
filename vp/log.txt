-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for NVDLA CMOD
-- NVDLA HW VERSION = nvdla_os_initial
-- NVDLA CMOD include directory = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/include
-- NVDLA CMOD library directory = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/lib
-- NVDLA CMOD library = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/lib/libnvdla_cmod.so
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for GreenLib
-- GreenLib include directory = /workspace/vp/libs/greenlib/greencontrol/include
-- GreenLib library directory = /workspace/vp/libs/greenlib/greenreg/lib
-- Searching for TLM2C
-- TLM2C include directory = /workspace/vp/libs/tlm2c/include
-- TLM2C library directory = /workspace/vp/libs/tlm2c.build
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for GreenLib
-- GreenLib include directory = /workspace/vp/libs/greenlib/greencontrol/include
-- GreenLib library directory = /workspace/vp/libs/greenlib/greenreg/lib
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for GreenLib
-- GreenLib include directory = /workspace/vp/libs/greenlib/greencontrol/include
-- GreenLib library directory = /workspace/vp/libs/greenlib/greenreg/lib
-- Searching for NVDLA CMOD
-- NVDLA HW VERSION = nvdla_os_initial
-- NVDLA CMOD include directory = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/include
-- NVDLA CMOD library directory = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/lib
-- NVDLA CMOD library = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/lib/libnvdla_cmod.so
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for GreenLib
-- GreenLib include directory = /workspace/vp/libs/greenlib/greencontrol/include
-- GreenLib library directory = /workspace/vp/libs/greenlib/greenreg/lib
-- Searching for SimpleCPU model.
-- SimpleCPU include directory = /workspace/vp/models/cpu/include
-- SimpleCPU library directory = /workspace/vp/models/cpu.build
-- Configuring done
-- Generating done
-- Build files have been written to: /workspace/vp
