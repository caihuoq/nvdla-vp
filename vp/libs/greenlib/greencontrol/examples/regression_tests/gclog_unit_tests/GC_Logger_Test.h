//   GreenControl framework
//
// LICENSETEXT
//
//   Copyright (C) 2009 : GreenSocs Ltd
// 	 http://www.greensocs.com/ , email: info@greensocs.com
//
//   Developed by :
//   
//   Michael Ruetz <m.ruetz@tu-bs.de>,
//   Christian Schroeder <schroeder@eis.cs.tu-bs.de>
//     Technical University of Braunschweig, Dept. E.I.S.
//     http://www.eis.cs.tu-bs.de
//
//
// The contents of this file are subject to the licensing terms specified
// in the file LICENSE. Please consult this file for restrictions and
// limitations that may apply.
// 
// ENDLICENSETEXT

// doxygen comments

#ifndef __GC_LOGGER_TEST_H__
#define __GC_LOGGER_TEST_H__

#include "greencontrol/log.h"
#include "TestCase.h"

// GCLOG_RETURN should be enabled
class GC_Logger_Test : public TestCase
{
public:
  GC_Logger_Test() {}
  virtual ~GC_Logger_Test(void) {}

  std::string identify()
  {
    return "GC_Logger_Test";
  }

  void setup()
  {
    pLog = new GC_Logger();
  }

  void destroy()
  {
    delete pLog;
  }

  // This is for testing the GC_Logger class
  void run()
  {
    DebugTextfile textOut("GC_Logger_Test.txt", false);
    ControlTransactionHandle ct, invCT;

    ct = getCT();
    invCT = getCT();
    invCT->set_mMetaData("This Transaction should not appear in any log!");

    // Test 1: Nothing should happen if there aren't any filters set.
    textOut.writePure("\nTest 1\nThere should be no output until test 2.");
    invCT->set_mSpecifier("Test 1 REQUEST");
    pLog->log(invCT);
    invCT->set_mSpecifier("Test 1 RESPONSE");
    pLog->logReturn(invCT);

    // Test 2: Only the first transaction should get logged, since the second has not changed.
    textOut.writePure("\nTest 2\nOnly the REQUEST should be printed.");
    pLog->setFilter(&textOut);
    ct->set_mSpecifier("Test 2 REQUEST");
    ct->set_mMetaData("This transaction should be the first in the log and not be followed by a RESPONSE");
    pLog->log(ct);
    pLog->logReturn(ct);
    ct->set_mMetaData("");

    // Test 3: Both transactions should get logged.
    textOut.writePure("\nTest 3\nThere should be a REQUEST followed by a RESPONSE.");
    ct->set_mSpecifier("Test 3 REQUEST");
    pLog->log(ct);
    ct->set_mSpecifier("Test 3 RESPONSE");
    pLog->logReturn(ct);

    // Test 4: Nothing should get logged, since the filters are deactivated.
    textOut.writePure("\nTest 4\nThe filters are deactivated now, so nothing should be printed.");
    pLog->setFilter(NULL);
    invCT->set_mSpecifier("Test 4 REQUEST");
    pLog->log(invCT);
    invCT->set_mSpecifier("Test 4 RESPONSE");
    pLog->logReturn(invCT);

    // Test 5: Only the RESPONSE should get logged, because only the returnFilter is set.
    textOut.writePure("\nTest 5\nOnly the RESPONSE should get logged.\n");
    ct->set_mSpecifier("Test 5 REQUEST");
    ct->set_mMetaData("This transaction should not get logged");
    pLog->setReturnFilter(&textOut);
    pLog->log(ct);
    ct->set_mSpecifier("Test 5 RESPONSE");
    ct->set_mMetaData("This RESPONSE should not be preceeded by a REQUEST. It should appear after Test 3.");
    pLog->logReturn(ct);

    // Test 6: Only the REQUEST should get logged, since the returnFilter is not set.
    textOut.writePure("\nTest 6\nOnly the REQUEST should get logged.");
    pLog->setFilter(&textOut);
    pLog->setReturnFilter(NULL);
    ct->set_mSpecifier("Test 6 REQUEST");
    ct->set_mMetaData("This transaction should not be followed by any other");
    pLog->log(ct);
    ct->set_mSpecifier("Test 6 RESPONSE");
    pLog->logReturn(ct);
  }

protected:
  GC_Logger *pLog;
};

#endif
