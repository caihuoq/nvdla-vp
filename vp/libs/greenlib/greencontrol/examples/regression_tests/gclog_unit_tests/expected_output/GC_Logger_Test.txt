Tuesday, 16 February 2010 - 13:06:06
Debug text output used with GC_Logger.
Starting log.


Test 1
There should be no output until test 2.

Test 2
Only the REQUEST should be printed.

  Transaction 0x93a2740:
    mHasChanged      = 0
    mService     (1) = 0 (NO_SERVICE)
    mTarget      (1) = 0
    mCmd         (1) = 0 (Warning: API/Plugin doesn't implement command_if!)
    mAnyPointer  (1) = 0
    mAnyPointer2 (1) = 0
    mAnyUint     (1) = 154805184
    mSpecifier   (1) = Test 2 REQUEST
    mValue       (1) = 
    mID          (1) = 0x93a1698 (Warning: API/Plugin doesn't implement command_if!)
    mMetaData    (1) = This transaction should be the first in the log and not be followed by a RESPONSE
    mError       (1) = 0
    mLogPointer  (1) = 0
    mCmdIf       (1) = 0
    mLogConfig   (1) = 0x93a2780 (hasChanged: 0)
    -> pNext     (0) = 0

Test 3
There should be a REQUEST followed by a RESPONSE.

  Transaction 0x93a2740:
    mHasChanged      = 0
    mService     (1) = 0 (NO_SERVICE)
    mTarget      (1) = 0
    mCmd         (1) = 0 (Warning: API/Plugin doesn't implement command_if!)
    mAnyPointer  (1) = 0
    mAnyPointer2 (1) = 0
    mAnyUint     (1) = 154805184
    mSpecifier   (1) = Test 3 REQUEST
    mValue       (1) = 
    mID          (1) = 0x93a1698 (Warning: API/Plugin doesn't implement command_if!)
    mMetaData    (1) = 
    mError       (1) = 0
    mLogPointer  (1) = 0
    mCmdIf       (1) = 0
    mLogConfig   (1) = 0x93a2780 (hasChanged: 0)
    -> pNext     (0) = 0
  Transaction 0x93a2740 returning:
    mHasChanged      = 1
    mService     (0) = 0 (NO_SERVICE)
    mTarget      (0) = 0
    mCmd         (0) = 0 (Warning: API/Plugin doesn't implement command_if!)
    mAnyPointer  (0) = 0
    mAnyPointer2 (0) = 0
    mAnyUint     (0) = 154805184
    mSpecifier   (1) = Test 3 RESPONSE
    mValue       (0) = 
    mID          (0) = 0x93a1698 (Warning: API/Plugin doesn't implement command_if!)
    mMetaData    (0) = 
    mError       (0) = 0
    mLogPointer  (0) = 0
    mCmdIf       (0) = 0
    mLogConfig   (0) = 0x93a2780 (hasChanged: 1)
    -> pNext     (0) = 0

Test 4
The filters are deactivated now, so nothing should be printed.

Test 5
Only the RESPONSE should get logged.

  Transaction 0x93a2740 returning:
    mHasChanged      = 1
    mService     (0) = 0 (NO_SERVICE)
    mTarget      (0) = 0
    mCmd         (0) = 0 (Warning: API/Plugin doesn't implement command_if!)
    mAnyPointer  (0) = 0
    mAnyPointer2 (0) = 0
    mAnyUint     (0) = 154805184
    mSpecifier   (1) = Test 5 RESPONSE
    mValue       (0) = 
    mID          (0) = 0x93a1698 (Warning: API/Plugin doesn't implement command_if!)
    mMetaData    (1) = This RESPONSE should not be preceeded by a REQUEST. It should appear after Test 3.
    mError       (0) = 0
    mLogPointer  (0) = 0
    mCmdIf       (0) = 0
    mLogConfig   (0) = 0x93a2780 (hasChanged: 1)
    -> pNext     (0) = 0

Test 6
Only the REQUEST should get logged.

  Transaction 0x93a2740:
    mHasChanged      = 0
    mService     (1) = 0 (NO_SERVICE)
    mTarget      (1) = 0
    mCmd         (1) = 0 (Warning: API/Plugin doesn't implement command_if!)
    mAnyPointer  (1) = 0
    mAnyPointer2 (1) = 0
    mAnyUint     (1) = 154805184
    mSpecifier   (1) = Test 6 REQUEST
    mValue       (1) = 
    mID          (1) = 0x93a1698 (Warning: API/Plugin doesn't implement command_if!)
    mMetaData    (1) = This transaction should not be followed by any other
    mError       (1) = 0
    mLogPointer  (1) = 0
    mCmdIf       (1) = 0
    mLogConfig   (1) = 0x93a2780 (hasChanged: 0)
    -> pNext     (0) = 0

Regular end of log.
Finished on Tuesday, 16 February 2010 - 13:06:06
Closing log.